lite/backends/nnadapter/nnadapter/src/driver/nvidia_tensorrt/engine.cc

```diff
@@ -131,10 +214,52 @@ int Program::BuildFromCache(core::Cache* cache) {
   return NNADAPTER_DEVICE_INTERNAL_ERROR;
 }
 
+int Program::CheckInputsAndOutputs(uint32_t input_count,
+                                   core::Argument* input_arguments,
+                                   uint32_t output_count,
+                                   core::Argument* output_arguments) {
+  // Check inputs
+  for (uint32_t i = 0; i < input_count; i++) {
+    // Get actual type
+    auto& arg = input_arguments[i];
+    NNAdapterOperandType type;
+    arg.access(arg.memory, &type);
+    // Check dimensions count
+    uint32_t count = type.dimensions.count;
+    auto& src_dimensions = input_types_[i].dimensions;
+    if (count != src_dimensions.count) {
+      return NNADAPTER_INVALID_DIMENSIONS;
+    }
+    // Check dimensions data
+    bool is_matched = true;
+    int32_t* data = type.dimensions.data;
+    int32_t* src_data = src_dimensions.data;
+    for (uint32_t j = 0; j < count; j++) {
+      if (data[j] != src_data[j]) {
+        is_matched = false;
+        break;
+      }
+    }
+    if (is_matched) continue;
+    // Check dynamic dymensions data
+    NNADAPTER_CHECK_EQ(src_dimensions.dynamic_count, 3U);
+    for (uint32_t j = 0; j < count; j++) {
+      if (data[j] < src_dimensions.dynamic_data[1][j] ||
+          data[j] > src_dimensions.dynamic_data[2][j]) {
+        return NNADAPTER_INVALID_DIMENSIONS;
+      }
+    }
+  }
+  return NNADAPTER_NO_ERROR;
+}
+
 int Program::Execute(uint32_t input_count,
                      core::Argument* input_arguments,
                      uint32_t output_count,
                      core::Argument* output_arguments) {
+  int ret = CheckInputsAndOutputs(
+      input_count, input_arguments, output_count, output_arguments);
+  if (ret != NNADAPTER_NO_ERROR) return ret;
   std::vector<void*> ptrs;
   for (auto& item : device_data_) {
     ptrs.push_back(item.get());
```

Comment:
for (auto& device_buffer : device_buffers_) {
    device_ptrs.push_back(device_buffer.get());
}

Reply:
done